magento/pwa-studio
# [PWA-638] My Account: Saved Payments: Remove/Delete

## Description

As a shopper, when logged in and viewing the Saved Payments page in My Account I want to be able to delete/remove saved payment methods. 

 

In-Scope: 

Create workflow for deleting/confirming the removal of a saved payment method in both desktop and mobile

 

Requirements: 

CTA for deleting a payment method 
Confirm/Approve 
Ensure proper API and security requirements

## Related Issue





* [[PWA-638](https://jira.corp.magento.com/browse/PWA-638)] My Account: Saved Payments: Remove/Delete

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Compare Saved Payment Method UI with mocks and Verify Remove functionalities  when -
2. User does not have any Saved Payment Method -  Should not be any Delete Icon. 
3. User has one Saved Payment Method - Delete Icon should be present.
4. User has 2 Saved Payment Methods - user should be able to Delete both payment methods.
5. User has Multiple Saved Payment Methods - User should be able to delete all payment methods.
6. Verify Delete confirmation buttons Cancel/Delete works as expected.
7. Verify when 2 or multiple payment methods are present and user deleted Default payment >> Does next payment method show Default label?
8. Verify functionalities work from both left drawer and Top right Account menu.
9. Verify Existing user Saved Payment Method gets displayed.
10. Verify after logout from Saved Payment method page user is taken to Home page.
11. Verify UI error when graphql returns error while adding/editing, 
12. Verify in all supported browsers and devices.


## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
